src/OrbitBase/include/OrbitBase/SpanUtils.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_SPAN_UTILS_H_
+#define ORBIT_BASE_SPAN_UTILS_H_
+
+#include <absl/types/span.h>
+
+namespace orbit_base {
+
+// Create spans of specified size from an input std::vector. Note that the last span can be smaller
+// than the specified size if the input vector size is not a multiple of the span size.
+//
+// Example: Splitting data in spans that can be processed in parallel tasks:
+//
+// void ProcessObjectsInParallel(std::vector<Object>& objects) {
+//   TaskGroup task_group;
+//   for (absl::Span<Object> span : CreateSpansOfSize(objects, 1024)) {
+//     task_group.AddTask([span]() {
+//       for (Object& object : span) {
+//         Process(object);
+//       }
+//     });
+//   }
+// }
+//
+template <typename T>
+[[nodiscard]] inline std::vector<absl::Span<T>> CreateSpansOfSize(std::vector<T>& input_vector,
+                                                                  const size_t span_size) {
```

Comment:
I would prefer to make `input_vector` a pointer which makes it less likely that a user accidentally passes in a temporary object. But AFAIK we don't have a rule on that. So feel free to ignore.

Reply:
The constraint of using pointers for non-const parameters was recently removed in the style guide, I'll leave it as is. This function must be used with care anyways since the input vector must outlive the spans.